woocommerce/woocommerce-android
# [Scan to update inventory] Part 1 – Continuous scanning mode and bottom sheet

#### [Scan to update inventory] Part 1 – Continuous barcode scanning mode and bottom sheet UI
This is **Part 1** of the Scan to update inventory project
Part 2 – https://github.com/woocommerce/woocommerce-android/pull/10253 – "Quantity +1" button, snackbars and unit tests
Part 3 – https://github.com/woocommerce/woocommerce-android/pull/10278 – Manual quantity update
Part 4 – https://github.com/woocommerce/woocommerce-android/pull/10313 – "Show product details" button

### Description

Figma designs: w3JBReW1UJesCf02sj43kR-fi-775%3A34738

This PR introduces the following changes:
1. Updated barcode scanner – continuous scanning mode
2. Search product by SKU use case
3. Compose UI for "Quick inventory update" bottom sheet

Not included in this PR:
1. Unit tests
2. Navigation
3. Business logic for updating product
4. Manual stock quantity value input
5. Edge cases: product not stock-managed
6. Analytics events tracking

Next PR – https://github.com/woocommerce/woocommerce-android/pull/10253

### Testing instructions

1. Open product list
2. Click "barcode scanner" icon
3. Scan barcode with the SKU of existing product (you can use online barcode generator like [this one](https://barcode.tec-it.com/en/?data=ACGADO))
4. Verify product is found and the bottom sheet is opened

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/4527432/9705ba79-1148-44e3-ac0b-2a6d290f7ccd
